main.cpp

```diff
@@ -1,167 +1,251 @@
+'wchar_t'
+        - 'void' is a return type. you do not need to declare 3 variables of
+type 'void'. b) at the end of the function, call ignoreUnused once and pass all
+of your variables to it. see line 71 for an example
+
+3) Declare 10 free functions
+    each declaration should have a random number of parameters in the function
+parameter list. When naming your parameters, choose names that are relevant to
+the task implied by the function's name. remember: Name functions what they do
+        takeDogForWalk(int distanceInYards);
+    Name variables what they are.
+        int numStepsSinceStart;
+
+4) add { ignoreUnused( ); } after each declaration in place of the closing
+semicolon 5) pass each of your function parameters to the ignoreUnused function
+like you did in b) 6) if your function returns something other than void, add
+'return { };' at the end of it. 7) provide default values for an arbitrary
+number of parameters in the function parameter list.
+
+8) consult the coding style guide found in the Readme.MD and adjust the
+formatting of your 10 functions.  At this point, you might have something that
+looks like this: float someFunc2(bool yes, int bar=2) { ignoreUnused(yes, bar );
+return { }; } This does not conform with the coding standard for this course
+(check the Readme.MD) and needs to be corrected
+
+9) in the main function at the end:
+    for each of those functions declared,
+        a) write out how the function would look if called with correct
+arguments b) if the function returned anything, store it in a local variable via
+the 'auto' keyword. c) pass the local variables to ignoreUnused() as you did in
+2b) see main() for an example of this.
+
+10) click the [run] button.  Clear up any errors or warnings as best you can.
+ */
+
+// 2)
+void variableDeclarations()
+{
+    int numberOfSquirrels = 3;
+    int numberOfCats = 1;
+    int numberOfDogs = 2;
+
+    double magicNumber = 3.14;
+    double rootMeanSquare = 1.414;
+    double oddsOfRain = 0.5;
+
+    bool isRaining = false;
+    bool isSnowing = true;
+    bool isSunny = false;
+
+    float priceOfAdmission = 5.25f;
+    float priceOfLunch = 12.99f;
+    float priceOfCab = 20.75f;
+
+    char favouriteLetter = 'z';
+    char favouriteSymbol = '!';
+    char favouriteNumber = '4';
+
+    unsigned int planets = 8;
+    unsigned int provinces = 10;
+    unsigned int territories = 3;
+
+    ignoreUnused(numberOfSquirrels, numberOfCats, numberOfDogs, magicNumber, rootMeanSquare, oddsOfRain, isRaining, isSnowing, isSunny, priceOfAdmission, priceOfLunch, priceOfCab, favouriteLetter, favouriteNumber, favouriteSymbol, planets, provinces, territories);
+}
+
+/*
+ 10 functions
+ */
+
+/*
+ 1)
+ */
+int totalAnimals(int lions, int bears, double unicorns = 0.707)
+{
+    ignoreUnused(lions, bears, unicorns);
+    return {};
+}
+
+/*
+ 2)
+ */
+float costofOuting(float priceOfLunch, float priceOfAdmission, float priceOfCab)
```

Comment:
This function name does not conform to the programming guideline:
```
Name Things what they are.
Name Actions based on what they do
```

The name you chose reflects a property (noun) and does not contain a verb, which would reflect an action.
Functions perform work, so their names must contain a verb which describes the work being performed.
Examples of function names that contain verbs:
- `computeAValue()`
- `estimateDistanceTraveled()`
- `performDiagnostics()`

- [ ] revise